src/JustSaying/Messaging/Middleware/Handle/HandlerMiddlewareBuilder.cs

```diff
@@ -95,7 +95,7 @@ public HandlerMiddlewareBuilder Use(Func<HandleMessageMiddleware> middlewareFact
                 $"Handler middleware has already been specified for {typeof(TMessage).Name} on this queue.");
         }
 
-        _handlerMiddleware = new HandlerInvocationMiddleware<TMessage>(handlerResolver.ResolveHandler<TMessage>);
+        _handlerMiddleware = new HandlerResolveInvocationMiddleware<TMessage>(handlerResolver);
```

Comment:
Could this cause any behavioural changes to existing integrations?

It might be safer to have this be an opt-in feature, and then do one or the other based on that option's value.